### ISSUE TITLE
Fatal Exception: android.app.RemoteServiceException

### ISSUE THREAD

**Issue by smarttcon** [transistorsoft/flutter_background_geolocation#79]
## Your Environment
* Plugin version: 1.0.8
* Platform: Android
* OS version: 9.0
* Device manufacturer / model: Samsung Galaxy S9 and Note 9
* Plugin config:
```
bg.Config(
        reset: true,
        heartbeatInterval: 60,
        desiredAccuracy: bg.Config.DESIRED_ACCURACY_MEDIUMm
        distanceFilter: 500.0,
        stopOnTerminate: false,
        notification:
            bg.Notification(title: "Smarttcon", text: "Sincronizando dados.."),
        batchSync: true,
        autoSyncThreshold: 50,
        startOnBoot: true,
        forceReloadOnSchedule: true,
        debug: false,
        enableHeadless: true,
        autoSync: true,
        url: '#####',
        params: deviceParams,
        logLevel: bg.Config.LOG_LEVEL_VERBOSE,
        schedule: [
          '2-6 8:00-18:00',
        ]);
```
## Context
The app closes after 10 seconds normally

## Debug logs
``` 
Fatal Exception: android.app.RemoteServiceException: Context.startForegroundService() did not then call Service.startForeground(): ServiceRecord{683c9d3 u0 br.com.smarttcon.empresarial.mobile/com.transistorsoft.locationmanager.service.ActivityRecognitionService}
       at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1855)
       at android.os.Handler.dispatchMessage(Handler.java:106)
       at android.os.Looper.loop(Looper.java:214)
       at android.app.ActivityThread.main(ActivityThread.java:6981)
       at java.lang.reflect.Method.invoke(Method.java)
       at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:493)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1445)
```
![image](https://user-images.githubusercontent.com/46688108/59519651-b68a3a80-8e9e-11e9-8fcb-6489f818c4a6.png)

**Comment by christocracy** [comment]
Show me all the code you're using with the plugin.

**Comment by smarttcon** [comment]
```
bg.Config(
        reset: true,
        heartbeatInterval: 60,
        desiredAccuracy: bg.Config.DESIRED_ACCURACY_MEDIUMm
        distanceFilter: 500.0,
        stopOnTerminate: false,
        notification:
            bg.Notification(title: "Smarttcon", text: "Sincronizando dados.."),
        batchSync: true,
        autoSyncThreshold: 50,
        startOnBoot: true,
        forceReloadOnSchedule: true,
        debug: false,
        enableHeadless: true,
        autoSync: true,
        url: '#####',
        params: deviceParams,
        logLevel: bg.Config.LOG_LEVEL_VERBOSE,
        schedule: [
          '2-6 8:00-18:00',
        ]);

 bg.BackgroundGeolocation.onLocation((location) {
      print(location);
      SmarttconInjector.injector.locationUtils.subjectLocation
          .add(ll.LatLng(location.coords.latitude, location.coords.longitude));
    });

    bg.BackgroundGeolocation.onMotionChange((location) {
      print(location);
      SmarttconInjector.injector.locationUtils.subjectLocation
          .add(ll.LatLng(location.coords.latitude, location.coords.longitude));
    });

    final state = await bg.BackgroundGeolocation.ready(config);
    if (state.schedule.isNotEmpty) {
      bg.BackgroundGeolocation.startSchedule();
    } else {
      await bg.BackgroundGeolocation.setConfig(config);
      bg.BackgroundGeolocation.startSchedule();
    }
```

**Comment by christocracy** [comment]
- I don't see you assigning the variable `config` anywhere.  Update your previous post to show me where your assigning `bg.Config config = ...`
- This doesn't make sense:
```dart
final state = await bg.BackgroundGeolocation.ready(config);
    if (state.schedule.isNotEmpty) {
      bg.BackgroundGeolocation.startSchedule();
    } else {  // <-- Why?  The config has already been applied with #ready.
      await bg.BackgroundGeolocation.setConfig(config);
      bg.BackgroundGeolocation.startSchedule();
    }
```

**Comment by christocracy** [comment]
Show me logs where you reproduce this by observing  `$ adb logcat -s TSLocationManager`.

**Comment by smarttcon** [comment]
I can not capture the error with the plugged-in phone, it did not appear. I talked to the users, they told me what happens in network changes.

**Comment by christocracy** [comment]
Let me know how to reproduce "with network changes".

**Comment by smarttcon** [comment]
did not make a mistake, it was punctual. One thing I identified in the trace was the use of maps at the time of the error.

```
android.os.MessageQueue.nativePollOnce (MessageQueue.java)
android.os.MessageQueue.next (MessageQueue.java:326)
android.os.Looper.loop (Looper.java:181)
com.google.maps.api.android.lib6.gmm6.store.e.run (: com.google.android.gms.dynamite_mapsdynamite @ 17455081 @ 17.4.55 (100400-248795830): 96)
com.google.maps.api.android.lib6.gmm6.store.z.run (: com.google.android.gms.dynamite_mapsdynamite @ 17455081 @ 17.4.55 (100400-248795830): 11)
```

**Comment by smarttcon** [comment]
![image](https://user-images.githubusercontent.com/46688108/59523220-1b499300-8ea7-11e9-82a8-8dc97941ccc9.png)

**Comment by christocracy** [comment]
This plugin doesn't access any "Maps" apis.

**Comment by smarttcon** [comment]
exactly, but what he wanted to show was that the mistake happened the moment he opened the map.

**Comment by smarttcon** [comment]
More one error :( in prod (FATAL ERROR):

Fatal Exception: java.lang.RuntimeException: Unable to start service com.transistorsoft.locationmanager.service.HeartbeatService@719d436 with null: java.lang.NullPointerException: Attempt to invoke virtual method 'java.lang.String android.content.Intent.getAction()' on a null object reference
       at android.app.ActivityThread.handleServiceArgs(ActivityThread.java:3045)
       at android.app.ActivityThread.-wrap17(ActivityThread.java)
       at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1452)
       at android.os.Handler.dispatchMessage(Handler.java:102)
       at android.os.Looper.loop(Looper.java:148)
       at android.app.ActivityThread.main(ActivityThread.java:5443)
       at java.lang.reflect.Method.invoke(Method.java)
       at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:728)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:618)

**Comment by christocracy** [comment]
>java.lang.NullPointerException: Attempt to invoke virtual method 'java.lang.String android.content.Intent.getAction()' on a null object reference

This one I know why.

**Comment by christocracy** [comment]
`NullPointerException in HeartbeatService` is fixed in `1.0.9`.
